src/components/tl/mlx5/mcast/p2p/ucc_tl_mlx5_mcast_p2p.c

```diff
@@ -0,0 +1,148 @@
+
+    ucc_collective_finalize(obj->req);
+}
+
+void ucc_tl_mlx5_mcast_completion_cb(void* context, ucc_status_t status)
+{
+    ucc_tl_mlx5_mcast_p2p_completion_cb(context);
+}
+
+static inline ucc_status_t ucc_tl_mlx5_mcast_do_p2p_bcast_nb(void *buf, size_t len, ucc_rank_t my_team_rank, int dest,
+                                                             ucc_team_h team, ucc_context_h ctx,
+                                                             ucc_coll_callback_t *callback,
+                                                             ucc_coll_req_h *p2p_req, int is_send)
+{
+    ucc_status_t    status = UCC_OK;
+    ucc_coll_req_h  req    = NULL;
+    ucc_coll_args_t args;
+
+    args.mask              = UCC_COLL_ARGS_FIELD_ACTIVE_SET | UCC_COLL_ARGS_FIELD_CB;
+    args.coll_type         = UCC_COLL_TYPE_BCAST;
+    args.src.info.buffer   = buf;
+    args.src.info.count    = len;
+    args.src.info.datatype = UCC_DT_INT8;
+    args.src.info.mem_type = UCC_MEMORY_TYPE_HOST;
+    args.root              = is_send ? my_team_rank : dest;
+    args.cb.cb             = callback->cb;
+    args.cb.data           = callback->data;
+    args.active_set.size   = 2;
+    args.active_set.start  = my_team_rank;
+    args.active_set.stride = dest - my_team_rank;
+
+    status = ucc_collective_init(&args, &req, team);
+    if (ucc_unlikely(UCC_OK != status)) {
+        tl_error(ctx->lib, "nonblocking p2p init failed");
+        return status;
+    }
+
+    ((ucc_tl_mlx5_mcast_p2p_completion_obj_t *)args.cb.data)->req = req;
+
+    status = ucc_collective_post(req);
+    if (ucc_unlikely(UCC_OK != status)) {
+        tl_error(ctx->lib, "nonblocking p2p post failed");
+        return status;
+    }
+
+    *p2p_req = req;
+
+    return status;
+}
+
+static inline ucc_status_t do_send_nb(void *sbuf, size_t len, ucc_rank_t my_team_rank,
+                                      int dest, ucc_team_h team, ucc_context_h ctx,
+                                      ucc_coll_callback_t *callback, ucc_coll_req_h *req)
+{
+    return ucc_tl_mlx5_mcast_do_p2p_bcast_nb(sbuf, len, my_team_rank, dest,
+                                             team, ctx, callback, req, 1);
+}
+
+static inline ucc_status_t do_recv_nb(void *rbuf, size_t len, ucc_rank_t my_team_rank,
+                                      int dest, ucc_team_h team, ucc_context_h ctx,
+                                      ucc_coll_callback_t *callback, ucc_coll_req_h *req)
+{
+    return ucc_tl_mlx5_mcast_do_p2p_bcast_nb(rbuf, len, my_team_rank, dest,
+                                             team, ctx, callback, req, 0);
+}
+
+ucc_status_t ucc_tl_mlx5_mcast_p2p_progress(void *context)
+{
+    ucc_tl_mlx5_mcast_oob_p2p_context_t *oob_p2p_ctx;
+    ucc_context_h                        ctx ;
+    ucc_status_t                         status;
+
+    oob_p2p_ctx = (ucc_tl_mlx5_mcast_oob_p2p_context_t *)context;
+    ctx         = oob_p2p_ctx->base_ctx;
+
+    status = ucc_context_progress(ctx);
```

Comment:
why do you need progress function like this? user anyways is responsible to call ucc_context_progress

Reply:
I'm not sure I understand your point, how exactly you are going to use this progress function in test? Will you export symbols from MLX5 component for it? Do you already have such tests in next PRs?

Reply:
@Sergei-Lebedev No I do not have these tests and I have never implemented gtests before. At this point, I can remove this progress() wrapper from the code - anyway, we are not calling it in the code right now.